### DISCUSSION TITLE
Geo Search using Mongo $geoWithin and $geoIntersects

### DISCUSSION THREAD

**Discussion by lablancas** [payloadcms/payload#2984] [Feature Requests & Ideas]
In my use case, I have a relationship that is determined by GeoJSON point in one collection (e.g. Business) and GeoJSON polygon in another collection (e.g. Neighborhood).

When I want to find the businesses in a neighborhood, I use $geoWithin to see which businesses are within the neighborhood's polygon.

When I want to find the neighborhoods for a business, I use the $geoIntersects to see where the business point intersects with a neighborhood polygon.

```
interface Neighborhood {
  geojson: JSON // a GeoJSON Polygon
}

interface Business {
  geojson: JSON // a GeoJSON Point
}

// when looking for a neighborhood's businesses, the query is like this
db.collections.business.find({ 
  "geojson.geometry": {
        $geoWithin: { $geometry: neighborhood.geojson.geometry }
    }
});

db.collections.neighborhood.find({ 
  "geojson.geometry": {
        $geoIntersects: { $geometry: business.geojson }
    }
});
```

**Comment by lablancas** [comment]
I doubt this will be a priority so any advice on how to get this functionality will be greatly appreciated. I was hoping that I could build a custom resolver or use a field hook, but I have not found a straightforward way to do something like this. Thinking I may have to build my own database connection or expose this data through an external API endpoint...

**Comment by denolfe** [comment]
This functionality would have to be built into core, and I think this is something that we'd be willing to accept a PR for.

As of now, we support the `near` operator. In order to support `geoWithin` and `geoIntersects`, the logic for each of these would need to be added wherever near's is such as [this](https://github.com/payloadcms/payload/blob/4b60845c670a211b9052f48c7dd7793350f0e8c8/src/mongoose/sanitizeQueryValue.ts#L84) as well as tests. Here are the [near tests](https://github.com/payloadcms/payload/blob/master/test/collections-rest/int.spec.ts#L800).
